cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/cluster/SqlClusteredAgentScheduler.kt

```diff
@@ -241,7 +238,12 @@ class SqlClusteredAgentScheduler(
       .forEach {
         log.warn("Found zombie agent {}, removing it", it.key)
         activeAgents.remove(it.key, it.value)
-    }
+        // Cancel zombie futures interrupting their AgentExecutionAction threads
+        val f = activeAgentsFutures.remove(it.key)
+        if (f?.cancel(true) == false) {
+          log.warn("Unable to cancel agent execution future for {}", it.key)
```

Comment:
Minor point, but is it worth calling [isCancelled](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/Future.html#isCancelled()) and then only logging if that returns true?  And then logging an error instead of a warning?

Reply:
According to the javadoc of `cancel`:
```
After this method returns, subsequent calls to isDone will always return true. Subsequent calls to isCancelled will always return true if this method returned true.
```
So I think it's redundant to call `isCancelled` after `cancel`. I'll adjust the logging though to log in both cases.

Reply:
When `cancel` returns false, I was thinking calling `isCancelled` could still make sense -- because we'd learn if the task completed normally.  I don't think we need to log (nor call `isCancelled`) if `cancel` returns true.

Reply:
`isCancelled` doesn't indicate clearly if a task completed normally (only if it was cancelled before completing normally), in that case it's more useful to check `isDone`, so we know if the task is still left running after cleanupZombieAgents finish.

Reply:
Good point.